### ISSUE TITLE
Custom Integration Error

### ISSUE THREAD

**Issue by kitracer** [uvjustin/alarmdotcom#82]
Ever since v2 beta 1 I intermittently get this error. I just upgraded to beta 6 and still have this problem. Restarting HASS doesn't resolve this.

This error originated from a custom integration.

Logger: homeassistant.config_entries
Source: custom_components/alarmdotcom/controller.py:409
Integration: Alarm.com (documentation, issues)
First occurred: 11:52:40 AM (1 occurrences)
Last logged: 11:52:40 AM

Error setting up entry Alarm Grid, Inc: eric@cagewrx.com for alarmdotcom
Traceback (most recent call last):
  File "/config/custom_components/alarmdotcom/controller.py", line 409, in get_controller
    await controller.async_login()
  File "/usr/local/lib/python3.9/site-packages/pyalarmdotcomajax/__init__.py", line 147, in async_login
    await self._async_login_and_get_key()
  File "/usr/local/lib/python3.9/site-packages/pyalarmdotcomajax/__init__.py", line 670, in _async_login_and_get_key
    async with self._websession.post(
  File "/usr/local/lib/python3.9/site-packages/aiohttp/client.py", line 1138, in __aenter__
    self._resp = await self._coro
  File "/usr/local/lib/python3.9/site-packages/aiohttp/client.py", line 559, in _request
    await resp.start(conn)
  File "/usr/local/lib/python3.9/site-packages/aiohttp/client_reqrep.py", line 898, in start
    message, payload = await protocol.read()  # type: ignore[union-attr]
  File "/usr/local/lib/python3.9/site-packages/aiohttp/streams.py", line 616, in read
    await self._waiter
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 313, in async_setup
    result = await component.async_setup_entry(hass, self)  # type: ignore
  File "/config/custom_components/alarmdotcom/__init__.py", line 41, in async_setup_entry
    if not await controller.async_setup():
  File "/config/custom_components/alarmdotcom/controller.py", line 58, in async_setup
    self.api = await get_controller(
  File "/config/custom_components/alarmdotcom/controller.py", line 409, in get_controller
    await controller.async_login()
  File "/usr/local/lib/python3.9/site-packages/async_timeout/__init__.py", line 128, in __aexit__
    self._do_exit(exc_type)
  File "/usr/local/lib/python3.9/site-packages/async_timeout/__init__.py", line 203, in _do_exit
    raise asyncio.TimeoutError
asyncio.exceptions.TimeoutError

**Comment by elahd** [comment]
Would you mind running the integration for a few days with debugging enabled? This will provide more detailed logs next time you run into this problem. Add the following to configuration.yaml.

```yaml
logger:
  default: warn
  logs:
    custom_components.alarmdotcom: debug
    pyalarmdotcomajax: debug
```

**Comment by AritroSaha10** [comment]
Just started to get this error as well, here are my logs:

```
2022-02-03 19:28:21 DEBUG (MainThread) [custom_components.alarmdotcom] custom_components.alarmdotcom: Initializing Alarmdotcom from config entry.
2022-02-03 19:28:21 DEBUG (MainThread) [custom_components.alarmdotcom.controller] custom_components.alarmdotcom.controller: Setting up controller with config_entry: {'username': ‘<username>’, 'password': ‘<password>’, '2fa_cookie': None}
2022-02-03 19:28:21 DEBUG (MainThread) [pyalarmdotcomajax] Attempting to log in to Alarm.com
2022-02-03 19:28:25 DEBUG (MainThread) [pyalarmdotcomajax] Response status from Alarm.com: 200
2022-02-03 19:28:25 DEBUG (MainThread) [pyalarmdotcomajax] {'__VIEWSTATE': ‘<b64_string>’, '__VIEWSTATEGENERATOR': ‘…’, ‘…’: ‘…’, '__PREVIOUSPAGE': ‘…’}
2022-02-03 19:28:25 INFO (MainThread) [pyalarmdotcomajax] Attempting login to Alarm.com
2022-02-03 19:28:31 DEBUG (MainThread) [pyalarmdotcomajax] Got Provider: PROVIDER, User ID: <ID>
2022-02-03 19:28:31 DEBUG (MainThread) [pyalarmdotcomajax] Calling update on Alarm.com
2022-02-03 19:28:31 DEBUG (MainThread) [pyalarmdotcomajax.entities] Initialized ADCRelationshipType.SYSTEM (systems/system) My Home
2022-02-03 19:28:31 DEBUG (MainThread) [pyalarmdotcomajax.entities] Initialized ADCRelationshipType.PARTITION (devices/partition) partition1
2022-02-03 19:28:31 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry <PROVIDER>: <EMAIL> for alarmdotcom
Traceback (most recent call last):
File "/config/custom_components/alarmdotcom/controller.py", line 409, in get_controller
await controller.async_login()
File "/usr/local/lib/python3.9/site-packages/pyalarmdotcomajax/__init__.py", line 145, in async_login
await self.async_update()
File "/usr/local/lib/python3.9/site-packages/pyalarmdotcomajax/__init__.py", line 206, in async_update
await self._async_get_devices(ADCDeviceType.SENSOR, self.sensors)
File "/usr/local/lib/python3.9/site-packages/pyalarmdotcomajax/__init__.py", line 319, in _async_get_devices
entities = await self._async_get_items_and_subordinates(
File "/usr/local/lib/python3.9/site-packages/pyalarmdotcomajax/__init__.py", line 539, in _async_get_items_and_subordinates
async with self._websession.get(
File "/usr/local/lib/python3.9/site-packages/aiohttp/client.py", line 1138, in __aenter__
self._resp = await self._coro
File "/usr/local/lib/python3.9/site-packages/aiohttp/client.py", line 559, in _request
await resp.start(conn)
File "/usr/local/lib/python3.9/site-packages/aiohttp/client_reqrep.py", line 898, in start
message, payload = await protocol.read() # type: ignore[union-attr]
File "/usr/local/lib/python3.9/site-packages/aiohttp/streams.py", line 616, in read
await self._waiter
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/config_entries.py", line 327, in async_setup
result = await component.async_setup_entry(hass, self)
File "/config/custom_components/alarmdotcom/__init__.py", line 41, in async_setup_entry
if not await controller.async_setup():
File "/config/custom_components/alarmdotcom/controller.py", line 58, in async_setup
self.api = await get_controller(
File "/config/custom_components/alarmdotcom/controller.py", line 409, in get_controller
await controller.async_login()
File "/usr/local/lib/python3.9/site-packages/async_timeout/__init__.py", line 129, in __aexit__
self._do_exit(exc_type)
File "/usr/local/lib/python3.9/site-packages/async_timeout/__init__.py", line 212, in _do_exit
raise asyncio.TimeoutError
asyncio.exceptions.TimeoutError
```

**Comment by AritroSaha10** [comment]
Seems like removing the integration and then adding it again seems to fix it? Though, it might happen again.

**Comment by elahd** [comment]
Thanks. I'm working on this problem. For the time being, restarting Home Assistant should resolve the error.

**Comment by elahd** [comment]
This should be resolved in v0.2.0-beta.8. Try it out and reopen this issue if you're still running into trouble.
